### ISSUE TITLE
The NumericLiteralPrefix cop is abusing her authority

### ISSUE THREAD

**Issue by jaredbeck** [bbatsov/rubocop#3258]
In 0.41.1, the autocorrect for NumericLiteralPrefix has the following issue, as discussed in https://github.com/bbatsov/rubocop/issues/3123

## Input

```ruby
Date.new(2014, 01, 02)
Date.new(2014, 01, 13)
Date.new(2014, 10, 01)
```

I like to make the date-part columns line up.

## Expected behavior

The expected behavior is a diff with no changes.

```diff
Date.new(2014, 01, 02)
Date.new(2014, 01, 13)
Date.new(2014, 10, 01)
```

## Actual behavior

```diff
- Date.new(2014, 01, 02)
- Date.new(2014, 01, 13)
- Date.new(2014, 10, 01)
+ Date.new(2014, 0o1, 0o2)
+ Date.new(2014, 0o1, 13)
+ Date.new(2014, 10, 0o1)
```

## Steps to reproduce the problem

Paste the above into a file and run autocorrect on it.

## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.41.1 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15)
```

CC: @tejasbubane

**Comment by mikegee** [comment]
Your examples conveniently avoid problematic values. Try September: `Date.new(2014, 09, 01)`

**Comment by bbatsov** [comment]
Indeed. This is not a bug at all. Every numeric literal starting with 0 is interpreted as something in octal by Ruby.

**Comment by jaredbeck** [comment]
> Every numeric literal starting with 0 is interpreted as something in octal by Ruby.

Oooooo. I see, thanks. It's just coincidence that I haven't used `09`! I'll avoid the prefix in the future, obviously. Once again, linting helps me understand ruby better!

**Comment by aripollak** [comment]
Would it be reasonable to have the cop not auto-correct literals like `00` to `0o0` but to `0` instead?

**Comment by tejasbubane** [comment]
@aripollak There is a config option `EnforcedOctalStyle` which supports two values `zero_with_o` (which is the default) and `zero_only` which will auto-correct to `0` instead of `0o`. Refer: https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L759

**Comment by gmkohler** [comment]
Could there be a setting to _ignore_ 00?

**Comment by jonas054** [comment]
@gmkohler No, there isn't, see [the manual](https://github.com/bbatsov/rubocop/blob/v0.45.0/manual/cops_style.md#stylenumericliteralprefix).

Generally, I'm against ignoring and prefer enforcing, but in this case we can't, as I guess you've also noted, correct every `0` into `00`. Another configuration option would IMO complicate things a bit too much to be worth it.

**Comment by eveevans** [comment]
That's interesting, but... when I try to 
`Date.new(2014, 0o9, 0o1)`
it throw
`SyntaxError: (eval):2: Invalid octal digit
Date.new(2014, 0o9, 0o1)`

**Comment by jonas054** [comment]
That's right, you can only use digits 0-7 in octal numbers. So `9` in decimal equals `0o11` in octal.
